src/port1.0/portsandbox.tcl

```diff
@@ -82,6 +83,16 @@ proc portsandbox::set_profile {target} {
     if {${configure.ccache}} {
         lappend allow_dirs $ccache_dir
     }
+    if {${build.type} eq "xcode"} {
+        # whitelist directories used by Xcode tools
+        lappend allow_dirs $portdbpath/home/Library/Developer/Xcode
+        lappend allow_dirs $portdbpath/home/Library/Caches
+        lappend allow_dirs $portdbpath/home/Library/org.swift.swiftpm
+        lappend allow_dirs $portdbpath/home/.swiftpm
+        # explicitly whitelist source dir to work around problems building
+        # Xcode projects in-source
+        lappend allow_dirs $worksrcpath
```

Comment:
Does this really make a difference? `$worksrcpath` is a subdir of `$workpath` which is already allowed.

Reply:
I thought so, too, but it actually does. Without it, Xcode builds in my setup fail with "You don't have permission to save the file." I can provide a log showing this if you need proof.

Reply:
I believe you, it just makes you wonder how you can know in general when allowing a directory does and doesn't allow its subdirs.